### PR TITLE
adds growth serum, puts it in fly amanita

### PR DESCRIPTION
**Pull request by PKPenguin321**
:cl:
add: Recent studies have revealed that fly amanita can give you a huge trips.
/:cl:

Known bugs:
The resize to shrink the mob back down to regular size doesn't work at all, you keep getting bigger forever, send help

### CODE REVIEW

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 1]
```diff
@@ -1231,3 +1231,20 @@ datum/reagent/romerol
...
+	color = "#ff0000"//strong red. rgb 255, 0, 0
+
+/datum/reagent/growthserum/reaction_mob(mob/H, method=INGEST, reac_volume)
+	if(!H.reagents.has_reagent("growthserum", -1))//so you cant grow infinitely
+		H.resize = 2
+		H.visible_message("<span class='notice'>[H] suddenly doubles in size!</span>", "<span class='notice'>You're a big guy!</span>")
+	..()
+
+/datum/reagent/growthserum/on_mob_delete(mob/M)
+	M.resize = 0.5//doesnt do anythign why
```
Comment (PKPenguin321) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]: send help

Comment (Fox-McCloud) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]: This is called when a reagent is done metabolizing---just splashing it on someone would never have this be called.

Also, this should be typecasted for `mob/living/M` not all mobs.

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 2]
```diff
@@ -1231,3 +1231,20 @@ datum/reagent/romerol
...
 	new /obj/item/organ/body_egg/zombie_infection(H)
 	..()
+
+/datum/reagent/growthserum
+	name = "growth serum"
+	id = "growthserum"
+	description = "A commercial chemical designed to help older men in the bedroom."//not really it just makes you a giant
+	color = "#ff0000"//strong red. rgb 255, 0, 0
+
+/datum/reagent/growthserum/reaction_mob(mob/H, method=INGEST, reac_volume)
```
Comment (Fox-McCloud) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]: don't typecast this for all mobs, and certainly not as mob/H (standard is M).

I'd recommend just making this something that happens `on_mob_life`, but only actually happens on the first cycle, then the reagent deleted thing will work.

**Comment by Fox-McCloud** [comment]
Guess we'll see just how deep the rabbit hole goes, eh?

**Comment by jughu** [comment]
Fox no

File: code/modules/reagents/chemistry/reagents/other_reagents.dm (hunk 2 above)
Comment (PKPenguin321) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]: >only on the first cycle

no idea how to do this

Comment (Fox-McCloud) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]:
```
/datum/reagent/growthserum/on_mob_life(mob/living/M)
	if(current_cycle == 1)
		M.resize = 2
		M.visible_message("<span class='notice'>[M] suddenly doubles in size!</span>", "<span class='notice'>You're a big guy!</span>")
	..()	
```

Comment (RemieRichards) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]: for you

File: code/modules/reagents/chemistry/reagents/other_reagents.dm (hunk 1 above)
Comment (PKPenguin321) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]: what should i use instead, then?

Comment (Fox-McCloud) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]: `/datum/reagent/growthserum/on_mob_delete(mob/living/M)`

**Comment by lordpidey** [comment]
you'll also want to move it to on_mob_life, rather than reaction_mob

**Comment by optimumtact** [comment]
Finish this please.

**Comment by PKPenguin321** [comment]
I forgot all about this, might work on it either later today or tomorrow

**Comment by PKPenguin321** [comment]
Okay I lied I apologize, I'm not really in the ss13 coding mood, maybe next time I'm into it I'll reopen this

**Comment by lordpidey** [comment]
@PKPenguin321 mind if I work on it?

**Comment by PKPenguin321** [comment]
Not at all

**Pull request closed without merging**